test/e2e/apps/statefulset.go

```diff
@@ -248,6 +248,124 @@ var _ = SIGDescribe("StatefulSet", func() {
 			sst.WaitForRunningAndReady(*ss.Spec.Replicas, ss)
 		})
 
+		// This can't be Conformance yet because it depends on a default
+		// StorageClass and a dynamic provisioner.
+		It("should perform rolling updates and roll backs of template modifications with PVCs", func() {
+			By("Creating a new StatefulSet with PVCs")
+			*(ss.Spec.Replicas) = 3
+			sst := framework.NewStatefulSetTester(c)
+			sst.SetHttpProbe(ss)
+			ss, err := c.AppsV1().StatefulSets(ns).Create(ss)
+			Expect(err).NotTo(HaveOccurred())
+			sst.WaitForRunningAndReady(*ss.Spec.Replicas, ss)
+			ss = sst.WaitForStatus(ss)
+			currentRevision, updateRevision := ss.Status.CurrentRevision, ss.Status.UpdateRevision
+			Expect(currentRevision).To(Equal(updateRevision),
+				fmt.Sprintf("StatefulSet %s/%s created with update revision %s not equal to current revision %s",
+					ss.Namespace, ss.Name, updateRevision, currentRevision))
+			pods := sst.GetPodList(ss)
+			for i := range pods.Items {
+				Expect(pods.Items[i].Labels[apps.StatefulSetRevisionLabel]).To(Equal(currentRevision),
+					fmt.Sprintf("Pod %s/%s revision %s is not equal to current revision %s",
+						pods.Items[i].Namespace,
+						pods.Items[i].Name,
+						pods.Items[i].Labels[apps.StatefulSetRevisionLabel],
+						currentRevision))
+			}
+			sst.SortStatefulPods(pods)
+			err = sst.BreakPodHttpProbe(ss, &pods.Items[1])
```

Comment:
What's the purpose of breaking and restoring the http probe in the context of this test?

Reply:
No idea, this is taken from the very similar test below so I figured there must have been some reason for it.

Reply:
Since you're reusing the existing StatefulSet test but with PVCs, maybe it would be better to wrap the existing test in a common function that could be called by both tests.

Reply:
Sorry for the delay. Sure thing, testing it now. We'll see how bad my implementation is. :)